src/app/admin/adminApp.js

```diff
@@ -11,44 +11,39 @@ import {
 import { dataProvider } from "ra-data-simple-prisma";
 
 const authProvider = ({ signIn, signOut }) => ({
-  login: async (credentials) => {
-    console.log("login", credentials);
-    return signIn(credentials);
-  },
-  logout: () => {
-    console.log("logout");
-    return signOut();
-  },
-  checkError: ({ status }) => Promise.resolve(),
-  // checkError: ({ status }) => {
-  // if (status === 401 || status === 403) {
-  //   localStorage.removeItem("username");
-  //   return Promise.reject();
-  // }
-  // return Promise.resolve();
-  // },
-  // called when the user navigates to a new location, to check for authentication
-  checkAuth: () => Promise.resolve(),
-  // checkAuth: () => {
-  // return Promise.resolve();
-  // console.log("checkAuth");
-  // return localStorage.getItem("username")
-  //   ? Promise.resolve()
-  //   : Promise.reject();
-  // },
-  getPermissions: () => Promise.resolve(),
+    login: async (credentials) => {
+      await signIn(credentials);
```

Comment:
Why do you put await before call signIn function, if the callback function that you pass here is not async?

Reply:
Don't you think that you can simplify this part of code just by writing - 
`login: signIn,`

Reply:
The callback function is not async, however it returns a Promise (the function inside of it returns a promise). We're awaiting the singin and signout processes on the backend to finish before continuing on the frontend

Reply:
Ok. Leave it for now - we will simplify it later.
Please follow the second comment in this thread